src/ui/run-info-ui-handler.ts

```diff
@@ -391,12 +391,11 @@ export default class GameInfoUiHandler extends UiHandler {
 
       const pokeInfoTextContainer = this.scene.add.container(-85, 3.5);
       const textContainerFontSize = "34px";
-      const pSpecies = pokemon.species;
       const pNature = getNatureName(pokemon.nature);
-      const pName = pokemon.fusionSpecies ? pokemon.name : pSpecies.name;
+      const pName = pokemon.getNameToRender();
       const passiveLabel = (currentLanguage==="ko"||currentLanguage==="zh_CN"||currentLanguage==="zh_TW") ? i18next.t("starterSelectUiHandler:passive") : (i18next.t("starterSelectUiHandler:passive") as string).charAt(0);
       const abilityLabel = (currentLanguage==="ko"||currentLanguage==="zh_CN"||currentLanguage==="zh_TW") ? i18next.t("starterSelectUiHandler:ability") : (i18next.t("starterSelectUiHandler:ability") as string).charAt(0);
-      const pPassiveInfo = pokemon.passive ? `${passiveLabel+": "+allAbilities[starterPassiveAbilities[pSpecies.speciesId]].name}` : "";
+      const pPassiveInfo = pokemon.passive ? `${passiveLabel+": "+pokemon.getPassiveAbility().name}` : "";
```

Comment:
you are making it harder for yourself here. Use a template string to it's full potential:
```suggestion
      const pPassiveInfo = pokemon.passive ? `${passiveLabel}: ${pokemon.getPassiveAbility().name}` : "";
```